pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -125,6 +131,12 @@ type GCECloud struct {
 	useMetadataServer        bool
 	operationPollRateLimiter flowcontrol.RateLimiter
 	manager                  diskServiceManager
+	// nodeZones is a mapping from Zone to a sets.String of Node's names in the Zone
+	// it is updated by the nodeInformer
+	nodeZones          map[string]sets.String
+	nodeInformerSynced cache.InformerSynced
+	// Lock to access to node informers
+	nodeInformerLock sync.Mutex
```

Comment:
The name of this lock is misleading. It is protecting nodeZones, not nodeInformer.

Can you maybe rename it to nodeZonesLock or sth like that?